agent/php_environment.c

```diff
@@ -382,6 +382,61 @@ static void nr_php_gather_dispatcher_information(nrobj_t* env) {
 
   nro_set_hash_string(env, "Dispatcher", dstring);
 }
+void nr_php_process_environment_variable(const char* prefix,
+                                         const char* key,
+                                         const char* value,
+                                         nrobj_t* kv_hash) {
+  if ((NULL == prefix) || (NULL == kv_hash) || (NULL == key)) {
+    return;
+  }
+
+  if (nr_strlen(prefix) >= nr_strlen(key)) {
+    return;
+  }
+
+  if (0 == nr_strncmp(key, prefix, nr_strlen(prefix))) {
+    nro_set_hash_string(kv_hash, key, value);
+  }
+}
+static void nr_php_get_environment_variables() {
+  nrobj_t* parsed_key_val = NULL;
+  /*
+   * `environ` works for non-windows machines.
+   * Otherwise, we'd need to use *__p__environ() as well.
+   */
+  extern char** environ;
+
+  /*
+   * Initialize the metadata hash.  If there aren't any variables, we still need
+   * to send the empty hash.
+   */
+  NR_PHP_PROCESS_GLOBALS(metadata) = nro_new_hash();
+
+  /*
+   * If we are unable to get the environment don't try to parse it.
+   */
+  if (NULL == environ) {
+    nrl_warning(NRL_AGENT, "%s: Unable to access environmental variables.",
+                __func__);
+    return;
+  }
+
+  /*
+   * Iterate through the environment variables, searching for a single key or
+   * a set of keys with a prefix that the agent will use.
+   */
+  for (size_t i = 0; environ[i] != NULL; i++) {
+    parsed_key_val = nr_strsplit(environ[i], "=", 0);
+    if (NULL == parsed_key_val) {
+      return;
```

Comment:
Maybe worth a debug level warning if parsing environ fails?  Unlikely but would be hard to troubleshoot otherwise?

Reply:
Thanks looks good!